utils/visualizer/qviz/content_loader.py

```diff
@@ -2,250 +2,264 @@
-    :param revision_id: target revisionID
-    :return: a list of AddFile and a set of RemoveFile paths
+    Extract metadata from a Delta table by checking if the revision ID is in that table.
+    If that ID exists in the table, a revision key (e.g. qbeast.revision.1) is created and the metadata is returned.
+    It also calculates the number of symbols used to representate the visualization.
+    :param delta_table: a created delta table.
+    :param revision_id: Configuration entry for the target RevisionID. e.g. 1
+    :return: A dictionary with the metadata related to the revisionId, an integer representing the number of simbols used fot the visualization representation.
     """
-    add_files = list()
-    remove_paths = set()
-    for file in json_files:
-        (adds, removes) = load_single_log_file(file, revision_id)
-        add_files.extend(adds)
-        remove_paths = remove_paths.union(removes)
-
-    return add_files, remove_paths
 
-
-def load_single_log_file(file_path: str, revision_id: str) -> (list[dict], set[str]):
+    # We create a dataframe with all the add actions
+    df = delta_table.get_add_actions(True).to_pandas()
+    # List with all the revision ids
+    revision_ids = [rev_id.strip() for rev_id in df["tags.revision"]]  # Delete blank spaces in each id
+    revision_id = str(revision_id).strip()  # Delete blank spaces in revision_key
+
+    # Check if our table has a revision id that matches the given revision id
+    if revision_id in revision_ids:
+        print("Metadata found for the given RevisionID\n")
+        revision_key = f"qbeast.revision.{revision_id}"
+        metadata_str = delta_table.metadata().configuration[revision_key]
+        # The last command return a string, and we need a dict to access columnTransformers
+        metadata = json.loads(metadata_str)
+        # Calculate the number of symbols used for the visualization
+        dimension_count = len(metadata['columnTransformers'])
+        symbol_count = (dimension_count + 5) // 6
+        return metadata, symbol_count
+    else:
+        print(f"No metadata found for the given RevisionID.")
+        symbol_count = float('inf')
+        # Create a dataframe with all add actions
+        df = delta_table.get_add_actions(True).to_pandas()
+        # Convert all blocks JSON chains to a list of dictionaries
+        df['tags.blocks'] = df['tags.blocks'].apply(lambda x: json.loads(x) if isinstance(x, str) else x)
+        # Make sure that 'tags.blocks' is a list of dictionaries
+        exploded_df = df.explode('tags.blocks')
+        # Extract 'cubeId'
+        all_cube_ids = exploded_df['tags.blocks'].apply(lambda x: x['cubeId'] if isinstance(x, dict) and 'cubeId' in x else None).dropna().tolist()
+
+        for cube_id in all_cube_ids:
+            cube_encoding_size = len(cube_id)
+            if 0 < cube_encoding_size < symbol_count:
+                symbol_count = cube_encoding_size
```

Comment:
Why is `symbol_count` computed this way?

Reply:
This has been deleted, if we don't find the revision ID we will show a message saying the qbeast table doesn't exist.